src/Sylius/Bundle/CoreBundle/CatalogPromotion/CommandHandler/RemoveCatalogPromotionHandler.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\CatalogPromotion\CommandHandler;
+
+use Doctrine\ORM\EntityManager;
+use InvalidArgumentException;
+use Sylius\Bundle\CoreBundle\CatalogPromotion\Command\RemoveCatalogPromotion;
+use Sylius\Component\Core\Model\CatalogPromotion;
+use Sylius\Component\Promotion\Repository\CatalogPromotionRepositoryInterface;
+
+final class RemoveCatalogPromotionHandler
+{
+    public function __construct(
+        private CatalogPromotionRepositoryInterface $catalogPromotionRepository,
+        private EntityManager $entityManager,
+    ) {
+    }
+
+    public function __invoke(RemoveCatalogPromotion $command): void
+    {
+        /** @var CatalogPromotion|null $catalogPromotion */
+        $catalogPromotion = $this->catalogPromotionRepository->findOneBy(['code' => $command->code]);
```

Comment:
Should we also check the state of the catalog promotion here, so as not to remove the one that is active?